app/src/main/java/com/mapbox/maps/testapp/utils/NavigationSimulator.kt

```diff
@@ -42,146 +37,35 @@ class NavigationSimulator(
+    OverviewViewportStateOptions.Builder().geometry(routePoints)
+      .padding(EdgeInsets(100.0, 100.0, 100.0, 100.0)).build()
+  )
   private var gesturesEnabled = true
 
   init {
     initMapboxMap()
-  }
-
-  private val onIndicatorBearingChangedListener = OnIndicatorBearingChangedListener {
-    if (cameraFollowMode == CameraFollowMode.FOLLOW) {
-      mapView.getMapboxMap()
-        .setCamera(CameraOptions.Builder().pitch(CAMERA_TRACKING_PITCH).bearing(it).build())
-    }
-  }
-
-  private val onIndicatorPositionChangedListener = OnIndicatorPositionChangedListener {
-    if (cameraFollowMode == CameraFollowMode.FOLLOW) {
-      mapView.getMapboxMap()
-        .setCamera(CameraOptions.Builder().center(it).pitch(CAMERA_TRACKING_PITCH).build())
-      mapView.gestures.focalPoint = mapView.getMapboxMap().pixelForCoordinate(it)
-    }
-  }
-
-  private val onMoveListener = object : OnMoveListener {
-    override fun onMoveBegin(detector: MoveGestureDetector) {
-      if (cameraFollowMode != CameraFollowMode.NONE) {
-        onCameraTrackingDismissed()
-      }
-    }
-
-    override fun onMove(detector: MoveGestureDetector): Boolean {
-      return false
-    }
-
-    override fun onMoveEnd(detector: MoveGestureDetector) {}
+    viewportPlugin.defaultTransition = viewportPlugin.makeDefaultTransition(
+      DefaultViewportTransitionOptions.Builder().maxDurationMs(2000).build()
+    )
   }
 
   private val onMapClickListener = OnMapClickListener {
-    when (cameraFollowMode) {
-      CameraFollowMode.OVERVIEW -> {
-        setCameraToFollowPuck()
-      }
-      CameraFollowMode.FOLLOW -> {
-        setCameraToOverview(true)
-      }
-      CameraFollowMode.NONE -> {
-        setCameraToFollowPuck()
-      }
-    }
-    true
-  }
-
-  private fun setCameraToOverview(easeTo: Boolean = true) {
-    cameraFollowMode = CameraFollowMode.OVERVIEW
-    val camera = mapView.getMapboxMap().cameraForCoordinates(
-      locationProvider.route.coordinates(),
-      bearing = 0.0,
-      pitch = 0.0,
-      padding = EdgeInsets(100.0, 100.0, 100.0, 100.0)
-    )
-    if (easeTo) {
-      mapView.camera.apply {
-        easeTo(
-          cameraOptions {
-            pitch(camera.pitch!!)
-            bearing(camera.bearing!!)
-          },
-          MapAnimationOptions.mapAnimationOptions {
-            duration(EASE_TO_PUCK_DURATION_MS / 3)
-          }
-        )
-        val zoom = createZoomAnimator(
-          CameraAnimatorOptions.cameraAnimatorOptions(camera.zoom!!)
-        ) {
-          duration = EASE_TO_PUCK_DURATION_MS / 3
-          interpolator = AccelerateDecelerateInterpolator()
+    with(viewportPlugin.status) {
+      if (this is ViewportStatus.State) {
+        when (state) {
+          followingViewportState -> viewportPlugin.transitionTo(overviewViewportState)
+          overviewViewportState -> viewportPlugin.transitionTo(followingViewportState)
+          else -> viewportPlugin.transitionTo(followingViewportState)
```

Comment:
is `else` needed? it seems to be a sealed class

Reply:
state is not a sealed class, and it can be null, but maybe I can combine the overviewViewportState branch with else branch.

Reply:
combined the overviewViewportState branch with else branch.